src/arch/ofi/machine-onesided.C

```diff
@@ -142,8 +150,9 @@ void process_onesided_reg_and_put(struct fi_cq_tagged_entry *e, OFIRequest *req)
                        ncpyOpInfo->srcSize);
 
   ncpyOpInfo->isSrcRegistered = 1; // Set isSrcRegistered to 1 after registration
-
-  const char *rbuf  = (FI_MR_SCALABLE == context.mr_mode) ? 0 : (const char*)(ncpyOpInfo->destPtr);
+  // if FI_MR_VIRT_ADDR is not enabled, then we use offset from the
+  // registered address which would be a 0 offset in this case.
+  const char *rbuf  = ((FI_MR_SCALABLE == context.mr_mode) || ( FI_MR_VIRT_ADDR & context.mr_mode)==0) ? 0 : (const char*)(ncpyOpInfo->destPtr);
```

Comment:
This logic is repeated multiple times below- should pull it out into a static function or macro

Reply:
True.  This was a pattern inherited from the initial OFI implementation, so I left it alone as it is slightly ugly, but doesn't seem to have performance implications.

Reply:
I was thinking for readability/maintainability rather than performance, but up to you

Reply:
implemented DETERMINE_OFFSET macro for this